ndsi/sensor.py

```diff
@@ -36,6 +37,33 @@ def __init__(self, value=None):
     def __str__(self):
         return repr(self.value)
 
+
+@enum.unique
+class SensorType(enum.Enum):
+    HARDWARE = 'hardware'
+    VIDEO = 'video'
+    ANNOTATE = 'annotate'
+    GAZE = 'gaze'
+    IMU = 'imu'
+
+    @staticmethod
+    def supported_types() -> typing.Set['SensorType']:
+        return set(SensorType)
+
+    @staticmethod
+    def supported_sensor_type_from_str(sensor_type: str) -> typing.Optional['SensorType']:
+        try:
+            sensor_type = SensorType(sensor_type)
+        except ValueError:
+            return None
+        if sensor_type not in SensorType.supported_types():
```

Comment:
In the current implementation, `SensorType.supported_types()` returns all the `SensorType` cases. But it might not be true in the future, if/when we decide to deprecate a `sensor_type`.